docs/proposals/backup-restore.md

```diff
@@ -1,131 +1,258 @@
+
+#### Offline Snapshot for Air-Gapped Environments
+
+Create a snapshot of a sample image `registry-a.k8s.io/kube-apiserver:v1` and its referrer (e.g. signature) for an air-gapped environment:
+
+```console
+oras backup registry-a.k8s.io/kube-apiserver:v1 --include-referrers --output airgap-snapshot.tar
 ```
 
-The backup command will also have the ability to write output to a new compressed tar file where the contents are in a single oci-layout
-format. For example:
+Upon success, the output will be:
 
-```bash
-oras backup --output ./mirror.tgz  registry.k8s.io/kube-apiserver-arm64:v1.31.0 registry.k8s.io/kube-controller-manager-arm64:v1.31.0
+```console
+Pulled 1 descriptor(s) from registry-a.k8s.io/kube-apiserver:v1
+Found 1 linked referrer(s)
+Included referrers in backup: application/vnd.cncf.notary.signature
+Exported backup to airgap-snapshot.tar
+Backup completed: 2 artifact(s) written
 ```
 
-If the output specified is an existing directory, the output will be written in that directory in OCI layout format.
-If the output specified is an existing file, it will be overwritten.
-If the output specified is neither a file or a directory, file output is assumed.
-The file name does NOT need to end in `tgz`, but file output will be compressed tar file.
+Transfer the `.tar` file to the air-gapped system via a secured channel. Restore the tarball from local to another registry:
 
-### oras restore
+```console
+oras restore registry-b.k8s.io/kube-apiserver:v1 --input airgap-snapshot.tar
+```
+
+Upon success, the output will be:
+
+```console
+Loaded 2 artifact(s) from airgap-snapshot.tar
+Pushed image to registry-b.k8s.io/kube-apiserver:v1
+Pushed linked referrer: sha256:78833f9c870...
+Restore completed successfully
+```
+
+By default, the image and linked referrers are reliably restored to another registry with minimal steps. Users can use the `--exclude-referrers` flag to exclude linked referrers when using `oras restore`.
 
-The restore command will support reading a directory or a compressed tar file and writing the content to a remote registry.
+```console
+$ oras discover registry-b.k8s.io/kube-apiserver:v1
+registry-b.k8s.io/kube-apiserver@sha256:9081a6f83f4febf47369fc46b6f0f7683c7db243df5b43fc9defe51b0471a950
+└── application/vnd.cncf.notary.signature
+    └── sha256:78833f9c870d3b069cdd998cae33b935629399f24743e680ab3bebb90de76589
+        └── [annotations]
+            ├── org.opencontainers.image.created: "2025-06-10T20:25:53Z"
+            └── io.cncf.notary.x509chain.thumbprint#S256: '["xxxxxx"]'
+```
 
-#### Restore from a directory
+#### Backup Artifacts to a Directory and Restore to Another Registry
 
-An example of restoring from a directory:
+Backing up multiple artifacts to a directory:
+
+```console
+oras backup --output ./mirror registry.k8s.io/kube-apiserver-arm64:v1.31.0 registry.k8s.io/kube-controller-manager-arm64:v1.31.0
```

Comment:
Do we support backing up artifacts from different registries? If so, we should show it in the example.

Reply:
Backing up artifacts from different registries is not a common scenario so far and it is not listed in the scenario doc https://github.com/oras-project/oras/pull/1778. I don't think it's required for the 1st iteration right now. I am opening to hear suggestions and requirements to see if we could support it in future iterations.